controllers/idler/idler_controller.go

```diff
@@ -198,6 +181,56 @@ func (r *Reconciler) ensureIdling(ctx context.Context, idler *toolchainv1alpha1.
 	return r.updateStatusPods(ctx, idler, newStatusPods)
 }
 
+// Check if the pod belongs to a controller (Deployment, DeploymentConfig, etc) and scale it down to zero.
+// if it is a standalone pod, delete it.
+// Send notification if the deleted pod was managed by a controller, was a standalone pod that was not completed or was crashlooping
+func deletePodsAndCreateNotification(ctx context.Context, podLogger logr.Logger, pod corev1.Pod, isCrashlooping bool, r *Reconciler, idler *toolchainv1alpha1.Idler) error {
+	lctx := log.IntoContext(ctx, podLogger)
+	logger := log.FromContext(ctx)
+	var podreason string
+	podCondition := pod.Status.Conditions
+	for _, podCond := range podCondition {
+		if podCond.Type == "Ready" {
+			podreason = podCond.Reason
+		}
+	}
+	appType, appName, deletedByController, err := r.scaleControllerToZero(lctx, pod.ObjectMeta)
+	if err != nil {
+		return err
+	}
+	if !deletedByController { // Pod not managed by a controller. We can just delete the pod.
+		log.FromContext(ctx).Info("Deleting pod without controller")
+		if err := r.AllNamespacesClient.Delete(ctx, &pod); err != nil {
+			return err
+		}
+		podLogger.Info("Pod deleted")
+	}
+	if appName == "" {
+		appName = pod.Name
+		appType = "Pod"
+	}
+
+	// If a build pod is in "PodCompleted" status then it was not running so there's no reason to send an idler notification
+	if podreason != "PodCompleted" || deletedByController || isCrashlooping {
+		// By now either a pod has been deleted or scaled to zero by controller, idler Triggered notification should be sent
+		if err := r.createNotification(ctx, idler, appName, appType); err != nil {
+			logger.Error(err, "failed to create Notification")
+			if err = r.setStatusIdlerNotificationCreationFailed(ctx, idler, err.Error()); err != nil {
+				logger.Error(err, "failed to set status IdlerNotificationCreationFailed")
+			} // not returning error to continue tracking remaining pods
+		}
+	}
+	return nil
+}
+
+func isRestartingOften(podstatus corev1.PodStatus) (bool, int32) {
+	var restartCount int32
+	for _, status := range podstatus.ContainerStatuses {
+		restartCount += status.RestartCount
```

Comment:
So there can be multiple restart counters (one per container)? If it's the case then I guess we just need to pick the biggest number here. And not sum them up. Otherwise if there is a pod with 10 containers which all restart then just 3 pod restarts will cross the 30 restart threshold. Or I'm reading it wrong?

Reply:
yes, you're reading that right. Let me make the change to look for the max restartCount if multiple containers are crashlooping.
I will also include this in one of the test cases.

Reply:
done in [576b44c](https://github.com/codeready-toolchain/member-operator/pull/630/commits/576b44c40a3fbd763c7792142a84d8a245b83b23)